linera-base/src/identifiers.rs

```diff
@@ -1010,14 +1025,9 @@ impl FromStr for AccountOwner {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if let Some(owner) = s.strip_prefix("User:") {
-            Ok(AccountOwner::User(
-                Owner::from_str(owner).context("Getting Owner should not fail")?,
-            ))
-        } else if let Some(app_id) = s.strip_prefix("Application:") {
-            Ok(AccountOwner::Application(
-                UserApplicationId::from_str(app_id)
-                    .context("Getting UserApplicationId should not fail")?,
+        if let Some(owner) = s.strip_prefix("Address32:") {
+            Ok(AccountOwner::Address32(
+                CryptoHash::from_str(owner).context("Getting Owner should not fail")?,
             ))
         } else if s.strip_prefix("Chain").is_some() {
```

Comment:
`s == "0x0"` ?

Reply:
I'm gonna leave that for another PR where I'll add the `Reserved` variant to `Address`.